### ISSUE TITLE
Do not install typing when running on the latest Python version

### ISSUE THREAD

**Issue by NicolasLM** [zalando/connexion#457]
### Description

Connexion has a hard dependency on `typing` which is useless when running the latest version of Python.

### Expected behaviour

Installing Connexion does not install typing when the interpreter is Python 3.6.1.

### Actual behaviour

```
$ pip install connexion
...
Collecting typing>=3.6.1 (from connexion)
  Downloading typing-3.6.1.tar.gz (66kB)
...
```

### Steps to reproduce

Install Connexion with Python 3.6.1.

### Additional info:

Python 3.6.1 on MacOS
Connexion Version: 1.1.10

**Comment by hjacobs** [comment]
Sounds reasonable.

**Comment by NicolasLM** [comment]
Thank you for taking the time to address this problem, however the `typing` dependency is still installed on the latest interpreter version because the wheel contains:

```
$ grep typing connexion-1.1.11.dist-info/METADATA 
Requires-Dist: typing (>=3.6.1)
```

As far as I know, when installing a wheel the `setup.py` is not executed. `setup.py` is only run on the interpreter building the wheel. My guess is that the person who built the wheel had a Python version below 3.6.

**Comment by hjacobs** [comment]
@NicolasLM ok, but building with Python 3.6 would break the wheel for users of Python 3.5, right?

**Comment by NicolasLM** [comment]
I believe the proper way to provide dependencies depending on the interpreter is by using environment markers: https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies

The example of the `enum` is particularly relevant.

**Comment by NicolasLM** [comment]
And I just saw that you implemented that in #495. I confirm that it works perfectly, thanks!
